### PR TITLE
UCP: Reduce worker address for unified mode

### PR DESCRIPTION
**Pull request by brminich**
## What
Reduce worker address size when unified mode is enabled

## Why ?
Worker address (which needs to be exchanged at startup) reduced as follows on one of the test machines:
- From 590b to 248b with default parameters
- From 51b to 36b with dc_x tls and specific HCA

## How ?
Do not pack interface attributes in unified mode. It is enough to pass resource index of every tl iface, so that the peer could take attrs from its local tl iface with the same resource index.

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8569/ for details (Mellanox internal link).

File: src/ucp/core/ucp_ep.c [hunk 1]
```diff
@@ -1116,22 +1116,27 @@ void ucp_ep_config_init(ucp_worker_h worker, ucp_ep_config_t *config)
...
     if (rndv_max_bw > 0) {
         for (i = 0; (i < config->key.num_lanes) &&
                     (config->key.rma_bw_lanes[i] != UCP_NULL_LANE); ++i) {
-            lane                         = config->key.rma_bw_lanes[i];
-            rsc_index                    = config->key.lanes[lane].rsc_index;
-            config->tag.rndv.scale[lane] = worker->ifaces[rsc_index].attr.bandwidth / rndv_max_bw;
+            lane      = config->key.rma_bw_lanes[i];
+            rsc_index = config->key.lanes[lane].rsc_index;
+
+            if (rsc_index != UCP_NULL_RESOURCE) {
```
Comment (hoopoepg) [inline src/ucp/core/ucp_ep.c]: 👍

File: src/ucp/wireup/address.c [hunk 2]
```diff
@@ -384,22 +429,21 @@ static ucs_status_t ucp_address_do_pack(ucp_worker_h worker, ucp_ep_h ep,
             wiface     = ucp_worker_iface(worker, i);
             iface_attr = &wiface->attr;
 
+            if (!(iface_attr->cap.flags & UCT_IFACE_FLAG_CONNECT_TO_IFACE) &&
```
Comment (hoopoepg) [inline src/ucp/wireup/address.c]: minor: ucs_test_flags may be used to simplify

**Comment by brminich** [comment]
@artpol84, @karasevb, FYI

File: src/ucp/wireup/address.c [hunk 3]
```diff
@@ -72,6 +72,12 @@ static size_t ucp_address_string_packed_size(const char *s)
     return strlen(s) + 1;
 }
 
+static int ucp_address_iface_attr_size(ucp_worker_t *worker)
```
Comment (yosefe) [inline src/ucp/wireup/address.c]: return size_t

File: src/ucp/wireup/address.c (hunk 2 above)
Comment (yosefe) [reply src/ucp/wireup/address.c]: or maybe `if (!(cap_flags & (CONNECT_TO_IFACE|CONNECT_TO_EP)))`

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8595/ for details (Mellanox internal link).

**Comment by brminich** [comment]
OOM + jenkins exceptions

bot:mlx:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8598/ for details (Mellanox internal link).

**Comment by brminich** [comment]
@hoopoepg, @yosefe - fixed

File: src/ucp/wireup/address.c [hunk 4]
```diff
@@ -72,12 +72,18 @@ static size_t ucp_address_string_packed_size(const char *s)
...
-static int ucp_address_iface_attr_size(ucp_worker_t *worker)
+static size_t ucp_address_iface_attr_size(ucp_worker_t *worker)
 {
     return worker->context->config.ext.unified_mode ?
            sizeof(ucp_rsc_index_t) : sizeof(ucp_address_packed_iface_attr_t);
 }
 
+static int ucp_worker_iface_can_connect(uct_iface_attr_t *attrs)
+{
+    return (attrs->cap.flags &
```
Comment (yosefe) [inline src/ucp/wireup/address.c]: why need ?: operator
if really need to return 0/1 (which is not the case here BTW) can do !!

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8601/ for details (Mellanox internal link).

**Pull request merged**